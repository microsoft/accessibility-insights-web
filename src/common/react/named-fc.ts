// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

export type ReactFCWithDisplayName<P = {}> = React.FC<React.PropsWithChildren<P>> & {
    displayName: string;
};

export function NamedFC<P = {}>(
    displayName: string,
    component: React.FC<React.PropsWithChildren<P>>,
): ReactFCWithDisplayName<P> {
    component.displayName = displayName;

    return component as ReactFCWithDisplayName<P>;
}
