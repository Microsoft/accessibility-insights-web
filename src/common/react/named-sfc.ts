// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

export type ReactSFCWithDisplayName<P = {}> = React.SFC<P> & { displayName: string };

export function NamedSFC<P = {}>(displayName: string, component: React.SFC<P>): ReactSFCWithDisplayName<P> {
    component.displayName = displayName;

    return component as ReactSFCWithDisplayName<P>;
}
