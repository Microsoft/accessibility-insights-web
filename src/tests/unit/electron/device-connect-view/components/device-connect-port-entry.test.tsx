// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { shallow } from 'enzyme';
import * as React from 'react';
import {
    DeviceConnectPortEntry,
    DeviceConnectPortEntryProps,
} from '../../../../../electron/device-connect-view/components/device-connect-port-entry';

describe('DeviceConnectPortEntryTest', () => {
    test('render', () => {
        const props = {} as DeviceConnectPortEntryProps;
        const rendered = shallow(<DeviceConnectPortEntry {...props} />);

        expect(rendered.getElement()).toMatchSnapshot();
    });
});
