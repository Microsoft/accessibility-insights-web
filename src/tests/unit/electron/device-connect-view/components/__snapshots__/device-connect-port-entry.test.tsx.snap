// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceConnectPortEntryTest render 1`] = `
<div
  className="device-connect-port-entry"
>
  <h3>
    Android device port number
  </h3>
  <MaskedTextField
    className="port-number-field"
    mask="99999"
    maskChar=""
    maskFormat={
      Object {
        "*": /\\[a-zA-Z0-9\\]/,
        "9": /\\[0-9\\]/,
        "a": /\\[a-zA-Z\\]/,
      }
    }
    onChange={[Function]}
    placeholder="12345"
  />
  <Button
    className="button-validate-port"
    disabled={true}
    onClick={[Function]}
  >
    Validate port number
  </Button>
</div>
`;
